### PR TITLE
[Issue 415 Ready to be merged] LSTM with peephole Support

### PR DESCRIPTION
**Pull request by psyyz10**
## What changes were proposed in this pull request?

Make BigDL support LSTM

## How was this patch tested?

Unit tests

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/480 https://github.com/intel-analytics/BigDL/issues/415

### CODE REVIEW

**Comment by jason-dai** [comment]
Please split this PR into several small ones (e.g., fix for Sequntial, NarrowTable) for merge.

File: dl/src/main/scala/com/intel/analytics/bigdl/nn/TimeDistributedCriterion.scala [hunk 1]
```diff
@@ -30,7 +30,7 @@ import scala.reflect.ClassTag
  */
 
 class TimeDistributedCriterion[T : ClassTag](critrn : TensorCriterion[T])
-(implicit ev: TensorNumeric[T]) extends TensorCriterion[T] {
+  (implicit ev: TensorNumeric[T]) extends TensorCriterion[T] {
```
Comment (jason-dai) [inline dl/src/main/scala/com/intel/analytics/bigdl/nn/TimeDistributedCriterion.scala]: revert the change

File: dl/src/main/scala/com/intel/analytics/bigdl/nn/TimeDistributedCriterion.scala [hunk 2]
```diff
@@ -99,7 +99,7 @@ class TimeDistributedCriterion[T : ClassTag](critrn : TensorCriterion[T])
 object TimeDistributedCriterion {
   def apply[@specialized(Float, Double) T: ClassTag](
     critrn: TensorCriterion[T] = null)
-  (implicit ev: TensorNumeric[T]) : TimeDistributedCriterion[T] = {
+    (implicit ev: TensorNumeric[T]) : TimeDistributedCriterion[T] = {
```
Comment (jason-dai) [inline dl/src/main/scala/com/intel/analytics/bigdl/nn/TimeDistributedCriterion.scala]: revert the change

**Comment by psyyz10** [comment]
Done
